app/Controller/UsersController.php

```diff
@@ -1231,13 +1231,7 @@ private function _postlogin()
       ));
       $lastUserLogin = $user['User']['last_login'];
       unset($user['User']['password']);
-      $user['User']['action'] = 'login';
-      $user['User']['last_login'] = $this->Auth->user('current_login');
-      $user['User']['current_login'] = time();
-      $this->User->save($user['User'], true, array('id', 'last_login', 'current_login'));
-      if (empty($this->Auth->authenticate['Form']['passwordHasher']) && !empty($passwordToSave)) {
```

Comment:
what about this special treatment of passwords?  The updateLoginTimes() function does not cover this

Reply:
Yes, because this functionality is broken. It depends on `$passwordToSave` variable, that is not defined in that function.

Reply:
Yep, looks like this was broken when the login function was split in two with _postlogin(). We can clean quite a bit of code up, it's from a few years ago when we transitioned to bcrypt for the password hashing.